framework/CodeInterfaces/PHISICS/phisicsdata.py

```diff
@@ -0,0 +1,961 @@
+    self.decayLabelListMrtau = []
+    self.numDensityLabelListMrtau = []
+    decayListMrtau = []
+    with open(os.path.join(self.workingDir,self.mrtauCSVOutputMPI), 'r') as outfile:
+      for line in outfile:
+        if re.search(r'INDIVIDUAL DECAY HEAT BLOCK',line):
+          decayFlag = 1
+        if re.search(r'CUMULATIVE DECAY HEAT BLOCK',line):
+          breakFlag = 1
+        if decayFlag == 1 and breakFlag == 0:
+          line = re.sub(r'\n',r'',line)
+          decayLine = filter(None, line.split(' '))
+          if decayLine != []: stringIsFloatNumber = self.isFloatNumber(decayLine)
+          if stringIsFloatNumber is True and decayLine != []:
+            if (float(decayLine[0]) == float(matchedTimeSteps[timeStepIndex])):
+              for i in xrange (0,len(self.isotopeListMrtau)):
+                if int(self.perturbationNumber) == 1:
+                  self.numDensityLabelListMrtau.append('numDensity'+'|'+self.isotopeListMrtau[i])
+                  self.decayLabelListMrtau.append('decay'+'|'+self.isotopeListMrtau[i])
+                decayListMrtau.append(decayLine[i+1])
+          if breakFlag == 1:
+            break
+    return decayListMrtau
+  
+  def getRRlist(self,instantDict):
+    """
+      Puts the reaction rate labels and the reaction rate values in a list.
+      @ In, instantDict, dictionary
+      @ Out, rrNames, list, list of parameter labels that are in the reaction rate matrix
+      @ Out, rrValues, list, values relative to the labels, within the reaction rate matrix
+    """
+    rrNames = []
+    rrValues = []
+    if self.printSpatialRR is True:
+      for i in xrange(0,len(self.paramList)):
+        for j in xrange(1,int(self.Ngroups) + 1):
+          for k in xrange(1, int(self.Nregions) + 1):
+            if self.paramList[i] == 'Group':
+              pass
+            else:
+              rrNames.append(self.paramList[i]+'|gr'+str(j)+'|reg'+str(k))
+              rrValues.append(instantDict.get('reactionRateInfo').get(str(j)).get(str(k)).get(self.paramList[i]))
+    if self.printSpatialRR is False:
+      for i in xrange(0,len(self.paramList)):
+        if self.paramList[i]   == 'Group':
+          pass
+        elif self.paramList[i] == 'Region':
+          pass
+        else:
+          rrNames.append(self.paramList[i]+'|Total')
+          rrValues.append(instantDict.get('reactionRateInfo').get(self.paramList[i]))
+    if 'Group' in rrNames:
+      rrNames.remove('Group')
+    return rrNames, rrValues
+    
+  def writeCSV(self,instantDict,timeStepIndex,matchedTimeSteps,jobTitle):
+    """
+      Prints the INSTANT/MRTAU data in csv files.
+      @ In, InstantDict, dictionary, contains all the values collected from INSTANT output
+      @ In, timeStepIndex, integer, timestep number
+      @ In, matchedTimeSteps, list, list of time steps considered
+      @ In, jobTitle, string, job title parsed from INSTANT input
+      @ Out, None
+      
+    """
+    if self.paramList != []:
```

Comment:
No. isinstance

Reply:
I am not cheklcing if this is a list or not, i am checking if the list is not empty